x-pack/plugins/metrics_entities/server/services/install_transforms.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ElasticsearchClient } from 'kibana/server';
+
+import { Transforms } from '../modules/types';
+import type { Logger } from '../../../../../src/core/server';
+
+import { computeMappingId, computeTransformId, getTransformExists } from './utils';
+import { logTransformError } from './utils/log_transform_error';
+import { logTransformDebug } from './utils/log_transform_debug';
+
+interface CreateTransformOptions {
+  esClient: ElasticsearchClient;
+  transforms: Transforms[];
+  autoStart: boolean;
+  indices: string[];
+  frequency: string;
+  logger: Logger;
+  query: object;
+  docsPerSecond: number | null;
+  maxPageSearchSize: number;
+  sync: {
+    time: {
+      delay: string;
+      field: string;
+    };
+  };
+  prefix: string;
+  suffix: string;
+}
+
+export const installTransforms = async ({
+  autoStart,
+  esClient,
+  frequency,
+  indices,
+  docsPerSecond,
+  logger,
+  maxPageSearchSize,
+  prefix,
+  suffix,
+  transforms,
+  query,
+  sync,
+}: CreateTransformOptions): Promise<void> => {
+  for (const transform of transforms) {
+    // If the user doesn't define an explicit mapping in the module then use the default
+    // of the transform id as the mapping index.
+    const destIndex = transform?.dest?.index ?? transform.id;
```

Comment:
Safe accessor needed here? Destination index should always be available?

Reply:
It is optional in the definition (for now) on disk. If it's not there, it takes the index from the transform.id as the backup and then prefixes it with a dot.

So that should be correct.